File: public/examples/integrations/toolkits/x/lookup_single_user_by_username_example_llm_oai.py
import os
from openai import OpenAI

USER_ID = "you@example.com"
PROMPT = "Look up the user 'TryArcade' on X (Twitter)."
TOOL_NAME = "X.LookupSingleUserByUsername"

client = OpenAI(
    base_url="https://api.arcade-ai.com",
    api_key=os.environ.get("ARCADE_API_KEY"))

response = client.chat.completions.create(
    messages=[
        {"role": "user", "content": PROMPT},
    ],
    model="gpt-4o-mini",
    user=USER_ID,
    tools=[TOOL_NAME],
    tool_choice="generate",
)
print(response.choices[0].message.content)

File: public/examples/integrations/toolkits/x/lookup_single_user_by_username_example_llm_oai.js
import OpenAI from 'openai';

const USER_ID = 'you@example.com';
const PROMPT = "Look up the user 'TryArcade' on X (Twitter).";
const TOOL_NAME = 'X.LookupSingleUserByUsername';

const client = new OpenAI({
    baseURL: 'https://api.arcade.dev',
    apiKey: process.env.ARCADE_API_KEY
});

const response = await client.chat.completions.create({
    messages: [
        { role: 'user', content: PROMPT }
    ],
    model: 'gpt-4o-mini',
    user: USER_ID,
    tools: [TOOL_NAME],
    tool_choice: 'generate'
});

console.log(response.choices[0].message.content); 